### ISSUE TITLE
Bucket tasks per project; add tags to TaskRequest.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [luci/luci-py#133]
```
To get accountability, we currently have --user. This is not sufficient.

We need tags, basically assigning arbitrary values to the tasks so it is easier 
to search for them.

Action items:

- New entity property to store this new data:
https://code.google.com/p/swarming/source/browse/services/swarming/server/task_r
equest.py#221
tags = ndb.StringProperty(repeated=True)

- New swarming.py client code to accept tags at CLI, e.g.
https://code.google.com/p/swarming/source/browse/swarming.py?repo=client#872
parser..add_option('--tag', action='append', default=[]

- New swarming client API to accept this value. Likely blocked on issue 118.

- New task presentation to show these tags
https://code.google.com/p/swarming/source/browse/services/swarming/templates/use
r_task.html

- New search functionality to search for tasks with a tag
https://code.google.com/p/swarming/source/browse/services/swarming/templates/use
r_tasks.html
https://code.google.com/p/swarming/source/browse/services/swarming/handlers_fron
tend.py#380
```

Original issue reported on code.google.com by `maruel@chromium.org` on 6 Aug 2014 at 4:12
- Blocking: #144, #149
